### PR TITLE
Fix/con 287/failsafe resource disposal

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/CON-287
Fixes issue with asset resource disposal after uploading to the cloud storage 

Is the must for https://github.com/oat-sa/lib-generis-gcp/pull/4

### CODE REVIEW

File: model/fileManagement/FlySystemManagement.php [hunk 1]
```diff
@@ -44,7 +44,10 @@ public function storeFile($fileSource, $label)
         $filename = $this->getUniqueFilename($label);
         $stream = $fileSource instanceof File ? $fileSource->readStream() : fopen($fileSource, 'r');
         $this->getFileSystem()->writeStream($filename, $stream);
-        fclose($stream);
+
+        if (is_resource($stream)) {
```
Comment (gabrielfs7) [inline model/fileManagement/FlySystemManagement.php:48]: Why was it required only now?

**Review by gabrielfs7 — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/fileManagement/FlySystemManagement.php (hunk 1 above)
Comment (boajer) [reply model/fileManagement/FlySystemManagement.php:48]: GCP driver cleaning up resources internally, in destructors. But all other implementations (aws/local/etc) we were using before not so pedantic.

**Review by gabrielfs7 — APPROVED** [review]
LGTM

**Review by bartlomiejmarszal — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**